storage/innobase/row/row0merge.cc

```diff
@@ -4336,7 +4336,8 @@ row_merge_rename_tables_dict(
 	renamed is a single-table tablespace, which must be implicitly
 	renamed along with the table. */
 	if (err == DB_SUCCESS
-	    && old_table->space_id) {
+	    && old_table->space_id
+		&& old_table->space != NULL) {
```

Comment:
I would prefer an implicit comparison to `NULL` whenever possible. This should fit in a single line:
```c++
 	if (err == DB_SUCCESS && old_table->space_id && old_table->space) {
```